src/main/java/mate/academy/onlinebookstore/service/shoppingcart/CartServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+package mate.academy.onlinebookstore.service.shoppingcart;
+
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import mate.academy.onlinebookstore.dto.item.ItemAddRequestDto;
+import mate.academy.onlinebookstore.dto.item.ItemResponseDto;
+import mate.academy.onlinebookstore.dto.item.ItemUpdateRequestDto;
+import mate.academy.onlinebookstore.dto.shoppingcart.CartResponseDto;
+import mate.academy.onlinebookstore.exceptions.AccessDeniedException;
+import mate.academy.onlinebookstore.exceptions.EntityNotFoundException;
+import mate.academy.onlinebookstore.mapper.CartMapper;
+import mate.academy.onlinebookstore.mapper.ItemMapper;
+import mate.academy.onlinebookstore.model.CartItem;
+import mate.academy.onlinebookstore.model.ShoppingCart;
+import mate.academy.onlinebookstore.model.User;
+import mate.academy.onlinebookstore.repository.item.ItemRepository;
+import mate.academy.onlinebookstore.repository.shoppingcart.CartRepository;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class CartServiceImpl implements CartService {
+    private final CartRepository cartRepository;
+    private final ItemRepository itemRepository;
+    private final CartMapper cartMapper;
+    private final ItemMapper itemMapper;
+
+    @Override
+    public CartResponseDto addBookToCart(ItemAddRequestDto requestDto, Long userId) {
+        ShoppingCart shoppingCart = retrieveShoppingCart(userId);
+        Optional<CartItem> itemOptional = itemRepository.findByBookIdAndShoppingCartId(
+                requestDto.getBookId(), shoppingCart.getId()
+        );
+
+        CartItem cartItem;
+        if (itemOptional.isEmpty()) {
+            cartItem = itemMapper.intoModel(requestDto);
+            cartItem.setShoppingCart(shoppingCart);
+        } else {
+            cartItem = itemOptional.get();
```

Comment:
Check java doc about methods `get()` and `orElseThrow()` in `Optional`
```suggestion
            cartItem = itemOptional.orElseThrow();
```

Reply:
The point is that we must provide logic if Optional is empty, so no exception is thrown.